.github/workflows/build-nightlies.yml

```diff
@@ -0,0 +1,55 @@
+name: Build nightlies
+
+on:
+  schedule:
+    # Every night at 4am ET
+    - cron: '0 8 * * *'
+  push: # TODO remove
+    branches:
+      - patcon/67-build-server-actions
+
+jobs:
+  build:
+    runs-on: ubuntu-latest
+    env:
+      GHA_COMMITS_TODAY: false
+      GHA_BRANCH: patcon/67-build-server-actions # TODO set to dev
+      # Use native docker command within docker-compose
+      COMPOSE_DOCKER_CLI_BUILD: 1
+      # Use buildkit to speed up docker command
+      DOCKER_BUILDKIT: 1
+    steps:
+      - name: Detect if commits in past day
+        run: |
+          # Default branch assumed
+          curl -sL https://api.github.com/repos/$GITHUB_REPOSITORY/commits?sha=${{ env.GHA_BRANCH }} \
```

Comment:
This is a little nutty; Is this the right way to do this? Seems like Github Actions would have a better way of responding to changes on a regular basis. I guess it's a little challenging if you don't want it to build a no-op every night.

Are we sure all this date/time/commit logic makes sense? Want to make sure cause of 4am ET cron time.

Reply:
GHA seems to keep core helper logic to a minimum. I think their idea is that ppl can build other features using new Actions and the GitHub API itself, instead of GitHub writing it. e.g., https://github.com/styfle/cancel-workflow-action

We can def do a no-op every night. Should only be 300 of 2000 minutes per month that we're using